### ISSUE TITLE
Valid usage of undefined

### ISSUE THREAD

**Issue by jbzdarkid** [eslint/eslint#14478]
**Tell us about your environment**
Javascript with no dependencies. Analysis is being done by codacy (https://app.codacy.com)

* **ESLint Version:** 7.24.0
* **Node Version:** unknown
* **npm Version:** unknown

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

**Please show your full configuration:** I do not have this information, it is being provided by codacy.

<details>
<summary>Configuration</summary>

```js

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
if ([undefined, 'top', 'bottom'].includes(cell.end)) {
```

```bash

```

**What did you expect to happen?**

`undefined` is flagged here as "Unexpected use of undefined."
However, replacing it with `null` does not actually work.
In this case, `cell` is an object, which may or may not include the `end` property.

**What actually happened? Please include the actual, raw output from ESLint.**

**Comment by jbzdarkid** [comment]
I don't have the ESLint output directly, since it's getting parsed by Codacy. I guess I could try to build a minimal repro, but I don't know if I care that much, since I'm just ignoring this rule now.

**Comment by aladdin-add** [comment]
sorry, we cannot help without enough info.

maybe you should report the issue to codacy's maintainers - if you are not directly using eslint.

**Comment by jbzdarkid** [comment]
Good suggestion, I'll try that.
